tls/s2n_client_extensions.c

```diff
@@ -429,3 +445,17 @@ static int s2n_recv_client_sct_list(struct s2n_connection *conn, struct s2n_stuf
     /* Skip reading the extension, per RFC6962 (3.1.1) it SHOULD be empty anyway  */
     return 0;
 }
+
+
+static int s2n_recv_client_max_frag_len(struct s2n_connection *conn, struct s2n_stuffer *extension)
+{
+    uint8_t mfl_code;
+    GUARD(s2n_stuffer_read_uint8(extension, &mfl_code));
+    if (mfl_code >= S2N_TLS_MAX_FRAG_LEN_INVALID || mfl_code_to_length[mfl_code] > S2N_TLS_MAXIMUM_FRAGMENT_LENGTH) {
+        fprintf(stderr, "invalid Max Fragmentation Length requested, continuing TLS handshake with default length\n");
```

Comment:
I have added similar check for server opt-in API
-> ignore extension and return 0 (not failing TLS handshake)

But I don't like that the print statments showup when we run the unit tests now (I added a case to cover server not opted in for mfl)